plugins/logging/main.go

```diff
@@ -2,20 +2,61 @@ package main
 
 import (
 	"context"
+	"errors"
 	"time"
 
 	"github.com/gin-gonic/gin"
+	"github.com/rancher/opni/pkg/features"
 	"github.com/rancher/opni/pkg/plugins"
 	"github.com/rancher/opni/pkg/tracing"
 	"github.com/rancher/opni/pkg/util/waitctx"
 	"github.com/rancher/opni/plugins/logging/pkg/logging"
+	"k8s.io/client-go/rest"
 )
 
 func main() {
 	tracing.Configure("plugin_logging")
 	gin.SetMode(gin.ReleaseMode)
 	ctx, ca := context.WithCancel(waitctx.Background())
-	plugins.Serve(logging.Scheme(ctx))
-	ca()
+	for {
+		if err := runPlugin(ctx, ca); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func runPlugin(ctx context.Context, cancel context.CancelFunc) error {
```

Comment:
When the gateway reloads, it will terminate and re-launch all plugins, so you might not need to handle it here.

Reply:
OK cool, I wasn't sure if it would leave the plugins running so I handled it here as a fallback.  I will drop it from here.